### PR TITLE
Burn and restore node id using external flash of Wismote

### PR DESCRIPTION
**Pull request by sumanpanchal**

### CODE REVIEW

File: platform/wismote/apps/burn-nodeid.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+static struct etimer etimer;
+
+PROCESS(burn_process, "Burn node id");
+AUTOSTART_PROCESSES(&burn_process);
+/*---------------------------------------------------------------------------*/
+PROCESS_THREAD(burn_process, ev, data)
+{
+  PROCESS_BEGIN();
+
+  etimer_set(&etimer, 5*CLOCK_SECOND);
```
Comment (nvt) [inline platform/wismote/apps/burn-nodeid.c]: There should be spaces around the multiplication sign.

File: platform/wismote/contiki-wismote-main.c [hunk 2]
```diff
@@ -229,10 +229,12 @@ main(int argc, char **argv)
...
    */
 
-  node_id = NODE_ID;
-
   /* Restore node id if such has been stored in external mem */
-  //node_id_restore();
+  node_id_restore();
+
+  if(!node_id) {
+      node_id = NODE_ID;
```
Comment (nvt) [inline platform/wismote/contiki-wismote-main.c]: This expression is indented too far.

**Comment by nvt** [comment]
This looks OK, except for a couple of style issues that I pointed out above.

**Comment by sumanpanchal** [comment]
@nvt 
Thanks for the feedback. I have fixed the code style as per your suggestions.

**Comment by nvt** [comment]
Another thing that I just noticed: the xmem.c module is not built by default for this platform, but the unconditional node_id_restore() call introduces a dependency on it.

**Comment by sumanpanchal** [comment]
xmem.c module will build by default for Wismote.
Makefile.wismote:
```
ARCH=spi.c xmem.c i2c.c node-id.c sensors.c cfs-coffee.c sht15.c \
     cc2520.c cc2520-arch.c cc2520-arch-sfd.c \
     sky-sensors.c uip-ipchksum.c \
     uart1.c slip_uart1.c uart1-putchar.c
```

**Comment by nvt** [comment]
Yes, I must have been looking at the wrong version before. I think this is ready to merge now. :+1:

**Comment by sumanpanchal** [comment]
Thanks.

**Pull request merged**